dbms/src/Flash/Executor/DataStreamExecutor.cpp

```diff
@@ -110,4 +111,27 @@ RU DataStreamExecutor::collectRequestUnit()
     cpu_time_ns /= 2;
     return toRU(ceil(cpu_time_ns));
 }
+
+Block DataStreamExecutor::getSampleBlock() const
+{
+    return data_stream->getHeader();
+}
+
+BaseRuntimeStatistics DataStreamExecutor::getRuntimeStatistics() const
+{
+    BaseRuntimeStatistics runtime_statistics;
+    auto & dag_context = dagContext();
+    auto root_executor_id = dag_context.getRootExecutorId();
+    const auto & profile_streams_map = dag_context.getProfileStreamsMap();
+    auto it = profile_streams_map.find(root_executor_id);
+    if (it != profile_streams_map.end())
+    {
+        for (const auto & input_stream : it->second)
+        {
+            if (auto * p_stream = dynamic_cast<IProfilingBlockInputStream *>(input_stream.get()); p_stream)
```

Comment:
Will the build time not collected if the root executor is `join`?

Reply:
The build time will not be collected, but now the `join` must not be the root executor.

Reply:
Why `join` must not be the root executor? I think the root executor is only decided by the planner in TiDB? And even if `join` is not the root executor, the executors after join will not collection the build time?

Reply:
- For MPP, the root executor must be the exchange sender.
- For cop/batch cop, the root executor must not be join.

But you are right, the build time will be lost even if the root executor isn't join.
Update to collect the execution time of `DataStreamExecutor::data_stream` directly